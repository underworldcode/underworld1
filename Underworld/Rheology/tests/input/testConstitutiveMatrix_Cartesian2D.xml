<?xml version="1.0"?>
<!DOCTYPE StGermainData SYSTEM "stgermain.dtd">

<StGermainData xmlns="http://www.vpac.org/StGermain/XML_IO_Handler/Jun2003">

	<include> Underworld/BaseApps/LidDrivenFEM.xml </include>
	<struct name="components" mergeType="merge">
		<struct name="context">
			<param name="Type">UnderworldContext</param>
		</struct>
		<struct name="rheology">
			<param name="Type">MaterialViscosity</param>
			<param name="eta0">1.0</param>
		</struct>	
		<struct name="shape">
			<param name="Type">Everywhere</param>
		</struct>		
		<struct name="material">
			<param name="Type">RheologyMaterial</param>
			<param name="Shape">shape</param>
			<list name="Rheology">
				<param>rheology</param>
			</list>
		</struct>			
	</struct>
	
	<list name="plugins" mergeType="merge">
		<param>Underworld_testConstitutiveMatrix</param>
	</list>

	<!-- Simulation control -->
	<param name="outputPath"> ./output </param>
	
	<!-- Journal Control -->
	<param name="journal-level.info.Stg_ComponentFactory">2</param>
	
	<!-- Geometry & Mesh setup -->
	<param name="dim"> 2 </param>
	<param name="elementResI"> 6 </param>
	<param name="elementResJ"> 6 </param>
	<param name="elementResK"> 1 </param>
	<param name="shadowDepth"> 1 </param>
	<param name="minX"> 0.0 </param>
	<param name="minY"> 0.0 </param>
	<param name="minZ"> 0.0 </param>
	<param name="maxX"> 1.0 </param>
	<param name="maxY"> 1.0 </param>
	<param name="maxZ"> 1.0 </param>
	<param name="allowUnbalancing"> True </param>
 
</StGermainData>
