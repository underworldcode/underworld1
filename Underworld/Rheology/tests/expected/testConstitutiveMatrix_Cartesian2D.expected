Constitutive Matrix after initialisation.
Constitutive Matrix after func 'SetMatrixWithThrees'.
matrixData[ 0 ][ 0 ] = 3; 	matrixData[ 0 ][ 1 ] = 3; 	matrixData[ 0 ][ 2 ] = 3; 	
matrixData[ 1 ][ 0 ] = 3; 	matrixData[ 1 ][ 1 ] = 3; 	matrixData[ 1 ][ 2 ] = 3; 	
matrixData[ 2 ][ 0 ] = 3; 	matrixData[ 2 ][ 1 ] = 3; 	matrixData[ 2 ][ 2 ] = 3; 	
constitutiveMatrix->isDiagonal = False
strainRate - 
      1           3     
      3           2     
stress - 
     27          27     
     27          27     
Constitutive Matrix after func 'SetMatrixWithViscosity7'.
matrixData[ 0 ][ 0 ] = 14; 	matrixData[ 0 ][ 1 ] = 0; 	matrixData[ 0 ][ 2 ] = 0; 	
matrixData[ 1 ][ 0 ] = 0; 	matrixData[ 1 ][ 1 ] = 14; 	matrixData[ 1 ][ 2 ] = 0; 	
matrixData[ 2 ][ 0 ] = 0; 	matrixData[ 2 ][ 1 ] = 0; 	matrixData[ 2 ][ 2 ] = 7; 	
constitutiveMatrix->isDiagonal = False
Viscosity =    7
strainRate - 
      1           3     
      3           2     
stress - 
     14          42     
     42          28     
stress - 
     14          42     
     42          28     
Constitutive Matrix after func 'SetMatrixWithSecondViscosity2'.
matrixData[ 0 ][ 0 ] = 12.72; 	matrixData[ 0 ][ 1 ] = 1.28; 	matrixData[ 0 ][ 2 ] = -0.96; 	
matrixData[ 1 ][ 0 ] = 1.28; 	matrixData[ 1 ][ 1 ] = 12.72; 	matrixData[ 1 ][ 2 ] = 0.96; 	
matrixData[ 2 ][ 0 ] = -0.96; 	matrixData[ 2 ][ 1 ] = 0.96; 	matrixData[ 2 ][ 2 ] = 6.28; 	
constitutiveMatrix->isDiagonal = False
strainRate - 
      1           3     
      3           2     
stress - 
   9.52       38.64     
  38.64       32.48     
